aws-eks-addon/vpc_cni.tf

```
@@ -0,0 +1,10 @@
+resource "aws_eks_addon" "vpc_cni" {
+  cluster_name = eks_properties.name
+  addon_name = "vpc-cni"
+  addon_version = addon_properties.vpc_cni_version
+  resolve_conflicts_on_update = "PRESERVE"
+  
+  node_group_name = [
+    eks_properties.node_security_group_name
+  ]
+}
```

you can change the addon to be one file in the root directory with a for each on the addon properties list

```suggestion
resource "aws_eks_addon" "eks_addon" {
    for_each = var.addon_properties

  cluster_name = eks_properties.name
  addon_name = each.key
  addon_version = each.value
  resolve_conflicts_on_update = "PRESERVE"
}
```

---

fixed